tableauserverclient/models/flow_run_item.py

```diff
@@ -2,50 +2,54 @@
 from ..datetime_helpers import parse_datetime
 import itertools
 
+from typing import Dict, List, Optional, Type, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from datetime import datetime
 
 class FlowRunItem(object):
     def __init__(self) -> None:
-        self._id=None
-        self._flow_id=None
-        self._status=None
-        self._started_at=None
-        self._completed_at=None
-        self._progress=None
-        self._background_job_id=None
+        self._id: str = ''
```

Comment:
Why change this from `None` to `str`? It might be ok but I'm a little nervous about the ripple impact in places where we might be checking this against is `None`

Reply:
I'm on the fence on this one. A flow run is a bit different theoretically from some of the other item types. A user of the TSC library would not find themselves creating a `FlowRunItem` to then interact with the server, nor should the server produce `FlowRunItem`s without an id. Leaving it as None makes the most sense from an API uniformity standpoint, but making the id non-optional in a sense reinforces the idea that an end user wouldn't create a flow run by starting with the python object.

I can change it back and update the type hints accordingly if you disagree with the idea behind hard setting it to string.

Reply:
That makes sense, I think it's fine as you have it then